### ISSUE TITLE
How to use for github pages?

### ISSUE THREAD

**Issue by sell** [jasonlong/cayman-theme#25]
Hey, this is my first time publishing my site with github.

I am fairly new, and am curious on how to use your theme using it, i get a _config.yml file, but thats it. I get no index, nor any of your css files. I was able to figure out how to https://sell.github.io/ edit the name by adding a description in the yml file, but how do I get your styles and everything for when making a index.html file? do I just copy your css files? Sorry, I am new to github, and I know how to do web development, but never used github for a site.

**Comment by sell** [comment]
Like, I do not know how to add the buttons how you did to the yml file, and not sure if I create a index page, but when I do create a index page it erases everything.

**Comment by jasonlong** [comment]
:wave: Hi @sell. I see you have https://sell.github.io/ working with the Cayman theme. By default, your site will simply present your `README` with your selected theme. If you want to customize your site with different html files, etc. the trick is that you need to put these files on a branch called `gh-pages`. 

[This part](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site) of the docs talks about this special branch.

And [here's an example](https://github.com/jasonlong/geo_pattern/tree/gh-pages) of one of my sites that's using the `gh-pages` branch (note the difference as you toggle between `master` and `gh-pages` with the branch picker).

Hope that helps!

**Comment by sell** [comment]
Hi @jasonlong , I appreciate this. I will look into this right now, I appreciate you taking the time out to reply to me. My name is also jason, once again thanks again. I am pretty new to github, I had my account for awhile but was always inactive.

**Comment by sell** [comment]
Sorry to bug you again @jasonlong  , but I do not see the option for that. When I went to source all I see is this :
 "Source
Your GitHub Pages site is currently being built from the master branch. Learn more.
User pages must be built from the master branch." 

It doesn't give me a drop down menu, like it shows in the [Configuring a publishing source for your GitHub Pages site](https://help.github.com/en/github/working-with-github-pages/configuring-a-publishing-source-for-your-github-pages-site)

# Update

I was able to make the branch I believe I did not do it correctly, now how do I get the files as you have them? Like the css, and etc. Do I manually do it, sorry im uncertain. Here is the [repo](https://github.com/sell/sell.github.io) that I am using for the site. If you go on the repo you can see master and gh-pages, but when you go on gh-pages it shows the same files and does not differ. Not sure what else I was supposed to do.

**Comment by jasonlong** [comment]
No worries at all. So when you created the `gh-pages` branch, it just makes a copy of what you have on `master`.  You can now just overwrite the files on `gh-pages` with your html, css, etc. files and it won't effect anything about `master`.

To overwrite those files, you can check that branch out locally using GitHub Desktop or the command line 

```
git fetch
git checkout gh-pages
```

And then make your changes, commit, and push like you normally would.

You could also make the changes right on the website but choose the `gh-pages` branch from the dropdown menu and then adding/removing files there.

Let me know if you get it working.

**Comment by sell** [comment]
So, I added the index.html file to gh-pages, and I do not see what I put on the site. Sorry man, just not familiar with doing this. Also, since I am using your theme, how do I get the css file for your theme, because I really wanted to add the buttons to the header, where it shows my name. I know how to make buttons, but not sure how to add it to this, since it is only showing the readme. 

You're probably just reading this laughing right now.

**Comment by jasonlong** [comment]
No need to apologize and I'm definitely not laughing. This part of our product isn't very well polished, but the good news is that we have a team starting up development on some new features ASAP.

Let's try this: I'm going to open a pull request on your repo to try to lay down some basic files you need in that branch. I haven't done this myself in a while, so it'll be a good refresher. :)

**Comment by sell** [comment]
Thanks for that, I appreciate it! I merged it, and I will wait for it to update, it currently still shows the old readme as default. I really appreciate your support, just for your support I am going to start a subscription on github. 

It is also exciting for me that a staff contributed to my code!

**Comment by jasonlong** [comment]
Looks like your site is now showing the correct content! 🎉 

As a student, make sure you sign up for the [GitHub Student Developer Pack](https://education.github.com/pack) if you haven't already. Lots of great stuff in there.

I'm going to close this issue, but let me know if you need any other help.

**Comment by sell** [comment]
Sorry to comment, and re-open again but are you able to show me a screenshot of what you see on your end?

This is what it shows for me, still shows the old content I had in my readme and not how it was on your repo.
![Screenshot_2020-02-20 About Me](https://user-images.githubusercontent.com/34258275/74964623-bd7c2c00-53e1-11ea-9af5-8e7fc0143ee1.png)

**Comment by jasonlong** [comment]
Oh shoot, sorry yes that's what I'm seeing as well. I thought you had added that content, but that's not the case.

Try making a change to the `index.html` file on your `gh-pages` branch and seeing if that triggers a rebuild of your site. 🤞

**Comment by sell** [comment]
I tried and just made a change now, and lets see if it works. Also, thanks for recommending me to sign up for github student, I didn't know they had that!

Also, in the settings of the page this is what I see :

> Source
Your GitHub Pages site is currently being built from the master branch. Learn more.
User pages must be built from the master branch.

Not sure if that has anything to do with it.

**Comment by jasonlong** [comment]
Ahh yes, that's the problem. This is all overly confusing, but for user pages (like for your username) you apparently need to have your html, css, etc. in `master`. 

Since you already have a `master` branch, the easiest thing is probably to just delete it from https://github.com/sell/sell.github.io/branches

Then, you can create a new `master` branch with this popover:

![image](https://user-images.githubusercontent.com/6104/74966396-f7026680-53e4-11ea-8d5e-784bac1448ca.png)

After that, I'd probably make `master` the default branch by going to Settings > Branches.

**Comment by sell** [comment]
Thanks for the help man I appreciate it! When I signed up on the education, where we had to say how am I going to use github, I left a note saying thanks to you for helping me. 

I belive it is all working now! Thanks for the help, one day i'll be like you and be able to work from home... possibly. 

Also, whould I be able to add a readme, just to have a description of the page when people click on the repo, or would that interfere with the code?
